#20 - add pages for every tab

Wiktor-Wojciechowski opened the pull request:
This should close #20 
Added basic pages based on the older version of the app

Wiktor-Wojciechowski committed:
added home placeholder+route

Wiktor-Wojciechowski committed:
added a test link

Wiktor-Wojciechowski committed:
added navbar home rawlog table page placeholders; added temporary icon; added routes; added default layout; changed styling

Wiktor-Wojciechowski committed:
added repositories, authors placeholders; updated home and navbar pages

Wiktor-Wojciechowski committed:
added footer to defaultlayout; cleaned up home; cleaned up navbar

Wiktor-Wojciechowski committed:
added file input to home

Wiktor-Wojciechowski committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

Wiktor-Wojciechowski committed:
home cleanup

Wiktor-Wojciechowski committed:
moved bottom buttons from home to defaultlayout

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
changed script to script setup, removed export in defaultlayout

Wiktor-Wojciechowski committed:
changed table, testing data passing

Wiktor-Wojciechowski committed:
added repositories view

Wiktor-Wojciechowski committed:
added authors page

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
added table page

Wiktor-Wojciechowski committed:
changed default layout

Wiktor-Wojciechowski committed:
fixed sorting in table

Wiktor-Wojciechowski committed:
added pinia

Wiktor-Wojciechowski committed:
moved logs sample to logsStore

Wiktor-Wojciechowski committed:
laout fix in home

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
added functions to modify logs (need fixing)

Wiktor-Wojciechowski committed:
modified the store to make the pages update along the data

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
made table reactive to log changes

Wiktor-Wojciechowski committed:
made repositories and authors reactive to log  changes

Wiktor-Wojciechowski committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

Wiktor-Wojciechowski committed:
added active link indicator

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
layout changes, elements dis/appear depending on whether the logs are loaded or not"

Wiktor-Wojciechowski committed:
added icons

Wiktor-Wojciechowski committed:
added titles to pages

Wiktor-Wojciechowski committed:
fixed recursive rerenders in table page

Wiktor-Wojciechowski committed:
added icons, fixed icons not displaying on table page

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
fixed repositories not displaying colors

Wiktor-Wojciechowski committed:
fixed broken sorting on table page

Wiktor-Wojciechowski committed:
changed logsStore file type from js to ts

Wiktor-Wojciechowski committed:
lint cleanup

PiotrFedak commented on resources/js/Pages/DefaultLayout.vue:
```diff
@@ -0,0 +1,71 @@
+import Navbar from './Navbar.vue'
+import { ref } from 'vue'
+
+const sample = `Date,Product,SKU,Quantity,Unit Type,Price Per Unit ($),Multiplier,Owner,Repository Slug,Username,Actions Workflow,Notes
+2022-05-20,Actions,Compute - UBUNTU,2,minute,0.008,1.0,galaxy,trantor,octocat,.github/workflows/check-pr.yml,
+2022-05-20,Actions,Compute - UBUNTU,9,minute,0.008,1.0,galaxy,caprica,octocat,.github/workflows/check-pr.yml,
+2022-05-20,Actions,Compute - UBUNTU,8,minute,0.008,1.0,galaxy,mustafar,octocat,.github/workflows/test.yml,
+2022-05-20,Actions,Compute - UBUNTU,9,minute,0.008,1.0,galaxy,caprica,octocat,.github/workflows/lint.yml,
+2022-05-20,Actions,Compute - UBUNTU,3,minute,0.008,1.0,galaxy,trantor,octocat,.github/workflows/behat.yml,
+2022-05-20,Actions,Compute - UBUNTU,3,minute,0.008,1.0,galaxy,mustafar,octocat,.github/workflows/check-pr.yml,
+2022-05-20,Actions,Compute - UBUNTU,2,minute,0.008,1.0,galaxy,trantor,octocat,.github/workflows/behat.yml,
+2022-05-20,Actions,Compute - UBUNTU,15,minute,0.008,1.0,galaxy,caprica,octocat,.github/workflows/lint.yml,
+2022-05-20,Actions,Compute - WINDOWS,7,minute,0.016,1.0,galaxy,mustafar,octocat,.github/workflows/test-windows.yml,
+2022-05-20,Actions,Compute - UBUNTU,11,minute,0.008,1.0,galaxy,caprica,octocat,.github/workflows/check-pr.yml,
+2022-05-20,Actions,Compute - UBUNTU,2,minute,0.008,1.0,galaxy,mustafar,octocat,.github/workflows/test.yml,
+2022-05-20,Actions,Compute - UBUNTU,1,minute,0.008,1.0,galaxy,trantor,dependabot[bot],.github/workflows/check-pr.yml,
+2022-05-20,Actions,Compute - UBUNTU,7,minute,0.008,1.0,galaxy,trantor,octocat,.github/workflows/behat.yml,
+`
```
I think having a large const like sample in the <script setup> is not ideal. It can be moved to a separate module or a JSON file

Wiktor-Wojciechowski replied:
I've moved the sample data to a separate file

Wiktor-Wojciechowski committed:
In response to PR comments, moved sample data from DefaultLayout to a seperate json file

Wiktor-Wojciechowski committed:
lint cleanup

EwelinaSkrzypacz commented on resources/data/sampleData.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "sample": "Date,Product,SKU,Quantity,Unit Type,Price Per Unit ($),Multiplier,Owner,Repository Slug,Username,Actions Workflow,Notes\n2022-05-20,Actions,Compute - UBUNTU,2,minute,0.008,1.0,galaxy,trantor,octocat,.github/workflows/check-pr.yml,\n2022-05-20,Actions,Compute - UBUNTU,9,minute,0.008,1.0,galaxy,caprica,octocat,.github/workflows/check-pr.yml,\n2022-05-20,Actions,Compute - UBUNTU,8,min ...
+}
```
```suggestion
}

```

EwelinaSkrzypacz reviewed (commented):
Please take care of warnings

![image](https://github.com/blumilksoftware/gha-analyzer/assets/56546832/c4937acc-d6cb-4faa-b6e2-d106ee4f2696)

Wiktor-Wojciechowski committed:
fixed lint warnings

Wiktor-Wojciechowski committed:
Update resources/data/sampleData.json

Co-authored-by: Ewelina Skrzypacz <56546832+EwelinaSkrzypacz@users.noreply.github.com>

PiotrFedak commented on resources/js/Pages/Authors.vue:
```diff
@@ -0,0 +1,165 @@
+const logsStore = useLogsStore()
+const logs = computed(() => logsStore.getLogs)
+
+const colors = [
+  'bg-gray-400',
+  'bg-blue-400',
+  'bg-red-400',
+  'bg-green-400',
+  'bg-yellow-400',
+  'bg-indigo-400',
+  'bg-purple-400',
+  'bg-pink-400',
+] 
```
This const is repeating in many files, you could simply add your colors to tailwind.config.js 


look at this: https://tailwindcss.com/docs/configuration

PiotrFedak commented on resources/js/Pages/Repositories.vue:
```diff
@@ -0,0 +1,164 @@
+  'bg-indigo-400',
+  'bg-purple-400',
+  'bg-pink-400',
+] 
```
Same here

PiotrFedak commented on resources/js/Pages/Table.vue:
```diff
@@ -0,0 +1,208 @@
+  'bg-indigo-400',
+  'bg-purple-400',
+  'bg-pink-400',
+]
```
and here

Wiktor-Wojciechowski committed:
moved colors from pages to tailwind config; changed module.exports to export default in tailwind config

Wiktor-Wojciechowski committed:
Merge branch '#20-add-pages-for-every-tab' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

PiotrFedak commented on resources/js/Pages/Authors.vue:
```diff
@@ -0,0 +1,160 @@
+import { useLogsStore } from '@/Stores/logsStore'
+import { Head } from '@inertiajs/vue3'
+import resolveConfig from 'tailwindcss/resolveConfig'
+import tailwindConfig from '../../../tailwind.config.js'
```
If your tailwind Config file is configured correctly you don't need to import it on every file you want to use it. It's global file working everywhere.

PiotrFedak replied:
I'm talking about this : `import tailwindConfig from '../../../tailwind.config.js`

sorry for marking all of 4 lines

Wiktor-Wojciechowski replied:
How can I access the properties without importing the config?

PiotrFedak commented on resources/js/Pages/Repositories.vue:
```diff
@@ -0,0 +1,159 @@
+import { useLogsStore } from '@/Stores/logsStore'
+import { Head } from '@inertiajs/vue3'
+import resolveConfig from 'tailwindcss/resolveConfig'
+import tailwindConfig from '../../../tailwind.config.js'
```
I think this should be removed

PiotrFedak commented on resources/js/Pages/Table.vue:
```diff
@@ -0,0 +1,206 @@
+import { useLogsStore } from '@/Stores/logsStore'
+import { Head } from '@inertiajs/vue3'
+import resolveConfig from 'tailwindcss/resolveConfig'
+import tailwindConfig from '../../../tailwind.config.js'
```
same

Wiktor-Wojciechowski committed:
colors can now be fetched from api

Wiktor-Wojciechowski committed:
added logs controller

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
cleanup

Wiktor-Wojciechowski committed:
updated files and paths

Wiktor-Wojciechowski committed:
changed colors json, colors still not working

Wiktor-Wojciechowski committed:
added a colors service, added authors repositories table ccontrollers

Wiktor-Wojciechowski committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

PrabuckiDominik committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

AmonDeShir committed:
add workflow actor

- Create model and database migration for the workflow actor
- Add type annotations to models
- Handle skipped workflows in GetWorkflowJobsRequest

AmonDeShir committed:
fix code style

AmonDeShir committed:
Rewrite frontend

AmonDeShir committed:
remove unused dependencies

AmonDeShir committed:
remove pinia

kamilpiech97 commented on app/Services/ColorsService.php:
```diff
@@ -0,0 +1,29 @@
+            return $data;
+        }
+
+        throw new \Exception("File not found: $this->colorsFilePath");
```
```suggestion
        throw new Exception("File not found: $this->colorsFilePath");
```

kamilpiech97 commented on resources/css/app.css:
```diff
@@ -1,3 +1,5 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
```suggestion
```

kamilpiech97 commented on resources/js/Pages/Table.vue:
```diff
@@ -0,0 +1,110 @@
+import Repository from '@/Components/Repository.vue'
+import { withSort } from '@/Utils/sort'
+
+interface WorkflowRun {
+  id: number
+  date: number
+  organization: string
+  repository: string
+  repository_id: number
+  minutes: number
+  price_per_minute: number
+  total_price: number
+  workflow: string
+  os: string
+  actor: {
+    id: number
+    name: string
+    github_id: number
+    avatar_url: string
+  }
+}
```
Maybe we should move that into seprated js file. Table should be more generic.

kamilpiech97 replied:
:eyes:

kamilpiech97 commented on tailwind.config.js:
```diff
@@ -1,11 +1,19 @@
+    ],
     theme: {
-        extend: {},
+        extend: {
+
+        },
+
```
```suggestion
        extend: {},
```

AmonDeShir committed:
remove ColorsService

AmonDeShir committed:
fix code style

AmonDeShir committed:
move GitHub to separate component

AmonDeShir committed:
create table component

AmonDeShir committed:
remove empty line

kamilpiech97 commented:
@AmonDeShir conflicts

AmonDeShir committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

AmonDeShir committed:
move interfaces to a separate folder

kamilpiech97 commented on resources/js/Types/WorkflowRun.d.ts:
```diff
@@ -0,0 +1,18 @@
+    name: string
+    github_id: number
+    avatar_url: string
+  }
```
It should not be separated type?

kamilpiech97 commented on app/Http/Controllers/AuthorsController.php:
```diff
@@ -0,0 +1,45 @@
+                }
+            }
+
+            $data[] = [
+                "id" => $actor->id,
+                "name" => $actor->name,
+                "github_id" => $actor->github_id,
+                "avatar_url" => $actor->avatar_url,
+                "minutes" => $minutes,
+                "price" => $price,
+            ];
```
Laravel Resources should be useful here.

kamilpiech97 commented on app/Http/Controllers/AuthorsController.php:
```diff
@@ -0,0 +1,45 @@
+            foreach ($actor->workflowRuns as $run) {
+                foreach ($run->workflowJobs as $job) {
+                    $minutes += $job->minutes;
+                    $price += $job->minutes * $job->price_per_unit;
```
Create custom attribute for that in model.

kamilpiech97 commented on app/Http/Controllers/RepositoriesController.php:
```diff
@@ -0,0 +1,43 @@
+                }
+            }
+
+            $data[] = [
+                "id" => $repo->id,
+                "name" => $repo->name,
+                "organization" => $repo->organization->name,
+                "avatar_url" => $repo->organization->avatar_url,
+                "minutes" => $minutes,
+                "price" => $price,
```
Laravel Resources should be useful here.

kamilpiech97 commented on app/Http/Controllers/RepositoriesController.php:
```diff
@@ -0,0 +1,43 @@
+            foreach ($repo->workflowRuns as $run) {
+                foreach ($run->workflowJobs as $job) {
+                    $minutes += $job->minutes;
+                    $price += $job->minutes * $job->price_per_unit;
```
Same as above.

kamilpiech97 commented on app/Http/Controllers/TableController.php:
```diff
@@ -0,0 +1,46 @@
+        $organizations = Organization::with("repositories.workflowRuns.workflowJobs")->with("repositories.workflowRuns.workflowActor")->get();
+        $data = [];
+
+        foreach ($organizations as $org) {
+            foreach ($org->repositories as $repo) {
+                foreach ($repo->workflowRuns as $run) {
+                    foreach ($run->workflowJobs as $job) {
+                        $data[] = [
+                            "id" => $run->job,
+                            "date" => $run->github_created_at,
+                            "organization" => $org->name,
+                            "repository" => $repo->name,
+                            "repository_id" => $repo->id,
+                            "minutes" => $job->minutes,
+                            "price_per_minute" => $job->price_per_unit,
+                            "total_price" => $job->minutes * $job->price_per_unit,
+                            "workflow" => $run->name . " - " . $job->name,
+                            "os" => $job->runner_os,
+                            "actor" => $run->workflowActor,
+                        ];
+                    }
+                }
+            }
+        }
```
It looks ugly :face_in_clouds: 
Lest use Laravel Resource and work with relationships :eyes:

kamilpiech97 commented on app/Http/Controllers/TableController.php:
```diff
@@ -0,0 +1,46 @@
+
+    public function show(): Response
+    {
+        $organizations = Organization::with("repositories.workflowRuns.workflowJobs")->with("repositories.workflowRuns.workflowActor")->get();
```
```suggestion
        $organizations = Organization::with(["repositories.workflowRuns.workflowJobs", "repositories.workflowRuns.workflowActor"])->get();
```
It looks better, lets think about getting data from `WorkflowJob` model and add to query some with relationship queries. (to avoid 4 time foreach)

AmonDeShir committed:
move actor interface to a separate file

AmonDeShir committed:
Merge branch 'main' of https://github.com/blumilksoftware/gha-analyzer into #20-add-pages-for-every-tab

AmonDeShir committed:
reduce nested loops

AmonDeShir committed:
move minutes and price calculations to model

AmonDeShir committed:
move job price calculation and workflow name to model

AmonDeShir committed:
move data processing to resources

kamilpiech97 commented on app/Models/WorkflowJob.php:
```diff
@@ -43,4 +47,14 @@ public function workflowRun(): BelongsTo
+
+    protected function fullName(): Attribute
+    {
+        return Attribute::get(fn() => $this->workflowRun->name . " - " . $this->name);
```
Lets add return type to `fn() =>`.

AmonDeShir committed:
Add return type to lambdas
